### DISCUSSION TITLE
Some basic help installing

### DISCUSSION THREAD

**Discussion by lycaki** [syssi/esphome-jk-bms#216] [Q&A] [answered]
Hey everyone,

having some issues, can't quite get my head sorted with installing this.
I'm using Home Assistant and ESPHOME (installed into HA). I can get my esp2866 flashed with the start and see it. I'm not sure how to add the jk-bms github. I've never done that.

I understand creating and linking the pins to the rx,tx etc (have setup up 8266 temp sensors etc)... into the UART but I'm unsure how to progress to get that installed onto the device and then into home assistant.

I've never linked a github to an esphome into HA, so I'm completely confused :)

Any help would be appreciated.

**Comment by syssi** [comment, marked as answer]
It's less complicated as you think. The most important part is to add this section to your YAML:

```
external_components:
  - source: github://syssi/esphome-jk-bms@main
```

This will retrieve the external component from this repository and makes it available. Use one of the example YAMLs from this repository as starting point. That's all!

**Reply by lycaki** [reply]
Arrggghhh thank you so much, I had tried that but my syntax was wrong on the 'external_components:' (I was putting it like this  external_components_source: github://syssi/esphome-jk-bms@main' 

I was getting an error to compile it. That was making me think it was all wrong! 

I see now, just added your line code to the top - it connected and compiled and uploaded straight away!

Thanks a bunch! :)
